specifyweb/frontend/js_src/lib/components/Header/Notifications.tsx

```diff
@@ -85,11 +122,20 @@ export function Notifications({
             }
           )
         )
-        .then(({ data: notifications }) => {
+        .then(({ data: newNotifications }) => {
           if (destructorCalled) return undefined;
-          setNotifications(
-            notifications
-              .map(
+          setNotifications((existingNotifications) => {
+            const filteredNewNotifications = newNotifications.filter(
+              (newNotification) =>
+                !existingNotifications?.some(
+                  (existingNotification) =>
+                    existingNotification.messageId ===
+                    newNotification.message_id
```

Comment:
can you reorder this code?
first .map() over notifications to convert `message_id` to `messageId` and the rest
and only afterward do `.filter()` and `.some()` - to avoid having to use message_id more than absolutely necessary